projects/addon-charts/components/line-chart/line-chart.component.ts

```diff
@@ -99,6 +109,20 @@ export class TuiLineChartComponent {
 
     value: readonly TuiPoint[] = [];
 
+    hovered$ = this._hovered$.pipe(distinctUntilChanged(), tuiZoneOptimized(this.ngZone));
+
+    leftHoverAxes$ = this.hints$.pipe(map(values => !!values.length)).pipe(
+        switchMap(hasHints =>
+            (this.axes?.hovered$.asObservable() ?? EMPTY).pipe(
+                map(hovered => hasHints || hovered),
+            ),
+        ),
+        debounceTime(100),
+        filter(visible => !visible),
+        tap(() => this.onHovered(NaN)),
```

Comment:
I investigate problem a little bit.
`TuiLineDaysChartComponent` uses `TuiLineChartComponent`:
https://github.com/Tinkoff/taiga-ui/blob/0a2daff2f3edfc44089a8dd1ac9b4e2d3db6ff37/projects/addon-charts/components/line-days-chart/line-days-chart.template.html#L1

But bug is NOT reproducible only for `LineChart` (either the chart has `[tuiLineChartHint]` or it does not)!


It happens because of 2 buggy moments :harold-pain:

### Case 1. The chart doesn't have `[tuiLineChartHint]`
`LineChart` processes this case via
https://github.com/Tinkoff/taiga-ui/blob/0a2daff2f3edfc44089a8dd1ac9b4e2d3db6ff37/projects/addon-charts/components/line-chart/line-chart.component.ts#L163-L168

It doesn't work for `LineDaysChart` because `!this.hintDirective` is always `false`.
`LineDaysChart` always provide `this.hintDirective`:
https://github.com/Tinkoff/taiga-ui/blob/0a2daff2f3edfc44089a8dd1ac9b4e2d3db6ff37/projects/addon-charts/components/line-days-chart/line-days-chart.component.ts#L42-L47

To solve this problem we should just duplicate these lines inside `TuiLineDaysChartComponent`(`TuiLineDaysChartComponent` has its own not-fake `this.hintDirective`):
```ts
@HostListener('mouseleave')
onMouseLeave(): void {
    if (!this.hintDirective) {
        this.onHovered(NaN);
    }
}
```
I check it. It solves bug for non-hint case.

### Case 2. The chart has `[tuiLineChartHint]`
When `LineChart` has hint, it doesn't manage hover-state.
`[tuiLineChartHint]` does it.
https://github.com/Tinkoff/taiga-ui/blob/0a2daff2f3edfc44089a8dd1ac9b4e2d3db6ff37/projects/addon-charts/components/line-chart/line-chart-hint.directive.ts#L55-L65

But the problem that for `LineDaysChart` it always returns empty query:
https://github.com/Tinkoff/taiga-ui/blob/0a2daff2f3edfc44089a8dd1ac9b4e2d3db6ff37/projects/addon-charts/components/line-chart/line-chart-hint.directive.ts#L38-L39

(yep, `TuiLineChartComponent` is inside `LineDaysChart`, but it is too nested and not reachable via `ContentChildren`, even with `{descendant: true}`).

But we can query `LineDaysChart` here (via `@ContentChildren()`).
We should find proper way to query `LineChart`-s via `LineDaysChart`.